website_snippet_country_dropdown/views/snippets.xml

```diff
@@ -54,7 +54,7 @@
 <template id="country_dropdown" name="Country Dropdown">
     <t t-set="default_value" t-value="default_value or ''"/>
     <t t-set="countries" t-value="request.env['res.country'].search([])"/>
-    <t t-set="default_country" t-value="default_country or countries.search([('code', '=', default_value[:2])]) or website.company_id.country_id or countries[:1]"/>
+    <t t-set="default_country" t-value="default_country or countries.search([('code', '=', default_value[:2])]) or request.env.user.country_id or website.company_id.country_id or countries[:1]"/>
```

Comment:
OK then. A bit twisted, because it's very easy to include `..or (default_value and countries.search([('code', '=', default_value[:2])])) or...` for readability purposes, but it works as you say.